crates/libs/windows/src/core/generic_factory.rs

```diff
@@ -0,0 +1,28 @@
+use super::*;
+use std::mem::*;
+
+// A streamlined version of the IActivationFactory interface used by WinRT class factories used internally by the windows crate
+// to simplify code generation. Components should implement the `IActivationFactory` interface published by the windows crate.
+#[repr(transparent)]
+#[derive(Clone, PartialEq, Eq)]
+pub struct IGenericFactory(IUnknown);
+
+impl IGenericFactory {
+    pub fn ActivateInstance<I: Interface>(&self) -> Result<I> {
```

Comment:
I know this is an internal API but some documentation would be helpful. Even if it just points out that this is the `ActivateInstance` from `IActivationFactory` (and then link [here](https://docs.microsoft.com/en-us/windows/win32/api/activation/nf-activation-iactivationfactory-activateinstance))